src/field_theory/galois.lean

```diff
@@ -329,23 +329,28 @@ end
 
 variables {F} {E} {p : F[X]}
 
-lemma of_separable_splitting_field_aux [hFE : finite_dimensional F E]
-  [sp : p.is_splitting_field F E] (hp : p.separable) (K : intermediate_field F E) {x : E}
+lemma of_separable_splitting_field_aux
+  [hFE : finite_dimensional F E]
+  [sp : p.is_splitting_field F E] (hp : p.separable)
+  (K : Type*) [field K] [algebra F K] [algebra K E]
+  [is_scalar_tower F K E] [finite_dimensional F K]
```

Comment:
This should follow from `finite_dimensional F E` + `is_scalar_tower F K E` (compare [`intermediate_field.finite_dimensional_left`](https://leanprover-community.github.io/mathlib_docs/field_theory/intermediate_field.html#intermediate_field.finite_dimensional_left)), but `library_search` isn't finding it at the moment...
```suggestion
  [is_scalar_tower F K E]
```

Reply:
Inferring `[finite_dimensional F K]` from `finite_dimensional F E + is_scalar_tower F K E` isn't allowed because lean has no way to find `E`.

Reply:
The instance is needed in the theorem statement though in order to provide a `fintype` instance. This gets messy quite quickly; should we just take the `fintype` argument directly (since it carries data anyway?)

Reply:
> The instance is needed in the theorem statement though in order to provide a `fintype` instance. This gets messy quite quickly; should we just take the `fintype` argument directly (since it carries data anyway?)

That's a good idea actually, there are in fact already non-defeq `fintype` instances on `alg_hom`.

Reply:
Arguably none of this matters anyway; this is an auxiliary lemma